cmd/query/app/querysvc/query_service.go

```diff
@@ -125,6 +131,32 @@ func (qs QueryService) GetDependencies(ctx context.Context, endTs time.Time, loo
 	return qs.dependencyReader.GetDependencies(ctx, endTs, lookback)
 }
 
+// MetricsQueryEnabled returns whether if metric query capabilities are enabled, opted-in by setting the
+// METRICS_STORAGE_TYPE environment variable.
+func (qs QueryService) MetricsQueryEnabled() bool {
+	return qs.options.MetricsReader != nil
+}
+
+// GetLatencies is the queryService implementation of metricsstore.Reader.
+func (qs QueryService) GetLatencies(ctx context.Context, params *metricsstore.LatenciesQueryParameters) (*metrics.MetricFamily, error) {
+	return qs.options.MetricsReader.GetLatencies(ctx, params)
```

Comment:
I think you will want to test for nil

Reply:
As above, these nil checks are performed by the handlers so at this point, `MetricsReader` should never be nil. However, if we take the proposed alternative of introducing a `MetricsQueryDisabled` error, then we could perform the `nil` check here.